### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by xFrednet**
@rust-lang/clippy, @Jarcho, @dswij, @Alexendoo. Could someone review this? It should be pretty straight forward since it's just a sync. I think it's also fine if either one of @Jarcho, @dswij, @Alexendoo approves this, as these are usually not reviewed. I just want to make sure that I didn't break something obvious :upside_down_face: 

It should be enough to look at the merge commit :upside_down_face: 

changelog: none
changelog: move [`significant_drop_in_scrutinee`] to `suspicious`

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Comment by xFrednet** [comment]
dogfood was still broken, and I forgot to change a lint group

@bors r=Manishearth

**Comment by xFrednet** [comment]
And one more try ^^

@bors r=Manishearth

**Comment by xFrednet** [comment]
The CI seems to be hanging. The test work locally, so I'm guessing this is a GH problem :thinking:

**Review by Manishearth — APPROVED** [review]

File: clippy_lints/src/significant_drop_in_scrutinee.rs [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    /// call in a match scrutinee, its lifetime lasts until the end of the match block, which may
+    /// be surprising.
+    ///
+    /// For `Mutex`es this can lead to a deadlock. This happens when the match scrutinee uses a
+    /// function call that returns a `MutexGuard` and then tries to lock again in one of the match
+    /// arms. In that case the `MutexGuard` in the scrutinee will not be dropped until the end of
+    /// the match block and thus will not unlock.
+    ///
+    /// ### Example
+    /// ```rust
```
Comment (Alexendoo) [inline clippy_lints/src/significant_drop_in_scrutinee.rs:32]: The deadlock example is getting doctested 😄

```suggestion
    /// ```rust,ignore
```

Comment (xFrednet) [reply clippy_lints/src/significant_drop_in_scrutinee.rs:32]: :joy: That is just funny :joy: Maybe that's why the CI didn't finish :joy:

**Comment by xFrednet** [comment]
Syncing takes a bit of practice. ^^

Alright one more try :):

@bors r=Manishearth,Alexendoo

**Pull request merged**